lua/autorun/client/cl_cfc_restart_display.lua

```diff
@@ -0,0 +1,71 @@
+local finalTime
+local keyFrame = 0
+local isOpen = false
+local flashState = false
+local CFCRestart = {}
+surface.CreateFont( "CFC_CALIBRI_20PX_DEFAULT", {
+    font = "Calibri",
+    extended = false,
+    size = 40,
+    weight = 500
+})
+
+local function timeLeft()
+    return finalTime - SysTime()
+end
+
+local function restartUI()
+    timer.Create( "CFC_RESTART_FLASHTIMER", 1, 0, function()
+        flashState = not flashState
+    end )
+
+    isOpen = true
+    hook.Remove( "HUDPaint", "CFCDrawRestartAlert" )
+    local delay = net.ReadInt( 16 )
+    local curTime = SysTime()
+    finalTime = curTime + delay
+    hook.Add( "HUDPaint", "CFCDrawRestartAlert", function()
+
+        if isOpen then
+            keyFrame = math.Clamp( keyFrame + 5, 0, ScrW() )
+        else
+            keyFrame = math.Clamp( keyFrame - 10, 0, ScrW() )
+
+            if keyFrame <= 0 then
+                hook.Remove( "HUDPaint", "CFCDrawRestartAlert" )
+            end
+        end
+
+        surface.SetDrawColor( Color( 36, 54, 72 ) )
+        surface.PlaySound( "ambient/alarms/warningbell1.wav" )
+        surface.DrawRect( 0, 5, keyFrame, 50 )
+    
+        CFCRestart.urgencyTime = 30 -- extracting that 30 out from here, should probably be named something better
+        CFCRestart.defaultDrawColor = Color( 255, 255, 89 )
+
+        local drawColor = self.defaultDrawColor
```

Comment:
This suggestion was built on the idea of putting all these functions into the `CFCRestart` table, in a similar style to this:
`local function restartUI()` -> `function CFCRestart:restartUI()`
the use of the colon here is what causes `self` to be defined. In its current state, this will error, so you have two options.

You can either change all your function definitions to be in the same style as above, using the colon. You'd also need to change all calls to these functions to also be in the same style:
`clearUI()` -> `CFCRestart:clearUI()`

Or you can change these `self`s to `CFCRestart`. In this case, I would still recommend putting your functions into the `CFCRestart` table, so they're accessible outside the addon, but this time you'd only use a dot, rather than a colon:
Definitions: `local function clearUI()` -> `function CFCRestart.clearUI()`
Calls: `clearUI()` -> `CFCRestart.clearUI()`

For future reference, you should always try to fully understand suggested code before implementing it into your PR, else you lose understanding of your own PR.